create-prs-to-update-vcs-repositories/create_prs_to_update_vcs_repositories.py

```diff
@@ -176,29 +164,66 @@ def get_logger(verbose: int) -> logging.Logger:
     return logging.getLogger(__name__)
 
 
-def get_latest_tag(tags: list[str], current_version: str) -> Optional[str]:
+def get_latest_tag(tags: list[str], current_version: str, target_release: str) -> Optional[str]:
     '''
     Description:
-        Get the latest tag from the list of tags
+        Get the latest tag from the list of tags based on the specified target release type.
 
     Args:
-        tags (list[str]): a list of tags
-        current_version (str): the current version of the repository
+        tags (list[str]): A list of tags.
+        current_version (str): The current version of the repository.
+        target_release (str): The type of release to check for updates. Can be 'major', 'minor', 'patch', or 'any'.
+
+    Returns:
+        Optional[str]: The latest tag that matches the target release type, or None if not found.
+
+    Raises:
+        ValueError: If an invalid target_release is specified.
     '''
+
+    valid_releases = {'major', 'minor', 'patch', 'any'}
+    if target_release not in valid_releases:
+        raise ValueError(f"Invalid target_release '{target_release}'. Valid options are: {valid_releases}")
```

Comment:
To avoid code duplication, use `VALID_RELEASES` here.
```suggestion
    if target_release not in VALID_RELEASES:
        raise ValueError(f"Invalid target_release '{target_release}'. Valid options are: {VALID_RELEASES}")
```

Reply:
Thank you, I applied the fixes by [this commit](https://github.com/autowarefoundation/autoware-github-actions/pull/320/commits/6fb60c86534cb4e1f2a68db9df84bba14ba64438).